### PR TITLE
merge data and master

### PR DESCRIPTION
**Pull request by wenchajun**
### Description
[Describe what this change achieves.]
 
### Issues Resolved
[List any issues this PR will resolve. You should likely open an issue if one does not already exist.]
 
### Check List
- [ ] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [ ] Helm chart version bumped
- [ ] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: charts/opensearch/values.yaml [hunk 1]
```diff
@@ -343,173 +128,763 @@ securityConfig:
...
+  singleNode: false
+  # The service that non master groups will try to connect to when joining the cluster
+  # This should be set to clusterName + "-" + nodeGroup for your master group
+  masterService: "opensearch-cluster-master"
+  # OpenSearch roles that will be applied to this nodeGroup
+  # These will be set as environment variable "node.roles". E.g. node.roles=master,ingest,data,remote_cluster_client
+  roles:
+    - master
+
+  replicas: 3
```
Comment (benjaminhuo) [inline charts/opensearch/values.yaml:155]: The master's replicas can be 1 instead

File: charts/opensearch/values.yaml [hunk 2]
```diff
@@ -343,173 +128,763 @@ securityConfig:
...
-  #    spec:
-  #      minAvailable: 1
-  #      selector:
-  #        matchLabels:
-  #          {{- include "opensearch.selectorLabels" . | nindent 6 }}
+data:
+  name: "opensearch-cluster-data"
+  clusterName: "opensearch-cluster"
+  nodeGroup: "data"
+  serviceName: "opensearch-cluster-data"
```
Comment (benjaminhuo) [inline charts/opensearch/values.yaml:530]: setting data node service name to `opensearch-cluster-data` will be different with kse 3.5's OpenSearch

Comment (wenchajun) [inline charts/opensearch/values.yaml:530]: Here is the same as 3.5, is configured two serverice, one is master-service and one is data-service.

**Comment by benjaminhuo** [comment]
Can be closed?

**Pull request closed without merging**